Improvement: correct architecture elements and modify component template

RolandJentschETAS opened the pull request:
## 📌 Description
Correct architecture elements and modify component template accordingly

## 🚨 Impact Analysis
- [x] This change does not violate any tool requirements and is covered by existing tool requirements
- [x] This change does not violate any design decisions
- [ ] Otherwise I have created a ticket for new tool qualification

## ✅ Checklist
- [x] Added/updated documentation for new or changed features
- [x] Added/updated tests to cover the changes
- [x] Followed project coding standards and guidelines

masc2023 reviewed (changes requested):
Why is feature not updated too in this PR, as it is also introduced? Should there not be an relationship between static architectures and components?

RolandJentschETAS commented:
> Why is feature not updated too in this PR, as it is also introduced? Should there not be an relationship between static architectures and components?

I'm on the relation to the code from the safety audit findings. Feature can be added later (small PRs).

RolandJentschETAS commented:
> > > Why is feature not updated too in this PR, as it is also introduced? Should there not be an relationship between static architectures and components?
> > 
> > 
> > I'm on the relation to the code from the safety audit findings. Feature can be added later (small PRs).
> 
> Disagree her, you introduced already the feature type, see your PR, so why not reflecting it directly, and I see no relationship why you should only consider audit findings as argument against it.

Added feature now ...

RolandJentschETAS commented on process/folder_templates/modules/module_name/component_name/docs/architecture/index.rst:
```diff
@@ -70,6 +70,13 @@ Static Architecture
+   :security: YES
+   :safety: ASIL_B
+   :status: invalid
+   :implements: logic_arc_int__feature_name__interface_name
```
There  is only the wording that it should be defined. There is nothing about, what belongs to/ consists of what and what is the linking between them. The linking is defined in [gd_req__arch_build_blocks_corr](https://eclipse-score.github.io/process_description/main/process_areas/architecture_design/guidance/architecture_process_reqs.html#gd_req__arch_build_blocks_corr) and there is no arrow between comp and interface_op.

masc2023 replied:
Will be followed-up later as part of https://github.com/eclipse-score/score/pull/2346
